src/discord.js

```diff
@@ -124,6 +117,78 @@ export default class DiscordAdapter extends Adapter {
     return false;
   }
 
+  configureAdapterOperations() {
+    this.bot.emitter.on(AT.DISCIPLINE_USER_WARNING, this.whisperUser, this);
+    this.bot.emitter.on(AT.DISCIPLINE_USER_TEMPORARY, this.kickUser, this);
+    this.bot.emitter.on(AT.DISCIPLINE_USER_PERMANENT, this.banUser, this);
+    this.bot.emitter.on(AT.WHISPER_USER, this.whisperUser, this);
+  }
+
+  whisperUser(adapterName, options) {
+    if (!adapterName || adapterName === this.name) {
+      const adapterUserId = this.getAdapterUserIdById(options.userId);
+      if (adapterUserId) {
+        this.client.fetchUser(adapterUserId)
+          .then(user => {
+            user.sendMessage(options.messageText)
+            .catch(reason => {this.bot.log.warning(reason.response.text);});
+          });
+      }
+    }
+  }
+
+  kickUser(adapterName, options) {
```

Comment:
if you make this `async kickUser`, you can clean up the code below:

```
...
const user = await this.client.fetchUser(adapterUserId);
await user.sendMessage(options.messageText);
const members = await Promise.all(this.client.guilds.map(g.fetchMember));
members.forEach(m => m.kick);
```

wrap necessary functions in try/catch.